hack/make.sh

```diff
@@ -14,10 +14,21 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+# posix complaint
+# verified by https://www.shellcheck.net
+
 # Simple script to offer the ability to run the make directives in a Docker Container.
 # In other words you don't need a Go env on your system to run Make (build, test etc)
 # This script will just bind-mount the source directory into a container under the correct
 # GOPATH and handle all of the Go ENV stuff for you.  All you need is Docker
-docker run -i -v "$PWD":/go/src/k8s.io/cloud-provider-vsphere:z \
-	-w /go/src/k8s.io/cloud-provider-vsphere \
-	golang:1.10 make "$@"
+
+# When in an interactive terminal add the -t flag so Docker inherits
+# a pseudo TTY. Otherwords SIGINT does not work to kill the container
+# when running this script interactively.
+TERM_FLAGS="-i"
+echo "${-}" | grep -q i && TERM_FLAGS="${TERM_FLAGS}t"
+
+docker run ${TERM_FLAGS} \
+  -v "$(pwd)":/go/src/k8s.io/cloud-provider-vsphere:z \
+  -w /go/src/k8s.io/cloud-provider-vsphere \
+  golang:1.11 make "${@}"
```

Comment:
can we parametrize the go version somewhere? it's going to be easier to source it from a known location across all the scripts in the repo.

Reply:
Hi @frapposelli,

Is it used elsewhere? I kind of feel the version of Go should be fixed. We don't necessarily want it to be easy for people to produce binaries with different Go versions. I think it's good that it's fairly assured that the image is the same each time.